src/main_lv.cpp

```diff
@@ -96,28 +137,44 @@ int main(int argc, char * argv[])
   read_ini(argv[1], settings);
 
   get_setting_child(general_settings, settings, "general", true);
-  get_setting_from_ptree(topic_root, general_settings, "topic_root", (std::string)"");
+  get_setting_from_ptree(topic_root, general_settings, "topic_root", (std::string) "");
   get_setting_child(ratslam_settings, settings, "ratslam", true);
+
   lv = new ratslam::LocalViewMatch(ratslam_settings);
 
-  if (!ros::isInitialized())
+  // initialize ROS node
+  ros::init(argc, argv, "RatSLAMViewTemplate");
+
+  // setup public and private nodehandles
+  ros::NodeHandle node;
+  ros::NodeHandle priv_node("~");
+
+  // params
+  priv_node.param("lvm_save_period", lvm_save_period, lvm_save_period);
+  priv_node.param("lvm_file_path", lvm_file_path, lvm_file_path);
+
+  // try to load lvm
+  if (!load_lvm(lvm_file_path))
   {
-    ros::init(argc, argv, "RatSLAMViewTemplate");
+    ROS_WARN("Could not load LocalViewMatch from file \"%s\"", lvm_file_path.c_str());
   }
-  ros::NodeHandle node;
 
-  ros::Publisher pub_vt = node.advertise<ratslam_ros::ViewTemplate>(topic_root + "/LocalView/Template", 0);
+  // pubs
+  pub_vt = node.advertise<ratslam_ros::ViewTemplate>(topic_root + "/LocalView/Template", 0);
 
+  // image transport
   image_transport::ImageTransport it(node);
-  image_transport::Subscriber sub = it.subscribe(topic_root + "/camera/image", 0, boost::bind(image_callback, _1, &pub_vt));
+  image_transport::Subscriber sub = it.subscribe("image", 0, image_callback);
```

Comment:
Note that the bag files for the examples use camera. So this is okay here but the launch files should be modified to remap and have a namespace.